From: Mike Hommey <mh@glandium.org>
Date: Tue, 19 May 2015 20:50:08 +0900
Subject: Bug 1166243 - Remove build() function from js and xpc shells.

This patch is cherry-picked from the Iceweasel package. It removes some
functions completely whose only meanings was to add a timestamp to the
binary.
https://bugzilla.mozilla.org/show_bug.cgi?id=1166243
---
 mozilla/ipc/testshell/XPCShellEnvironment.cpp | 10 ----------
 mozilla/js/src/shell/js.cpp                   | 13 -------------
 mozilla/js/xpconnect/src/XPCShellImpl.cpp     | 10 ----------
 3 files changed, 33 deletions(-)

diff --git a/mozilla/ipc/testshell/XPCShellEnvironment.cpp b/mozilla/ipc/testshell/XPCShellEnvironment.cpp
index 4b79683..df990d9 100644
--- a/mozilla/ipc/testshell/XPCShellEnvironment.cpp
+++ b/mozilla/ipc/testshell/XPCShellEnvironment.cpp
@@ -198,15 +198,6 @@ Version(JSContext *cx,
 }
 
 static bool
-BuildDate(JSContext *cx, unsigned argc, JS::Value *vp)
-{
-    JS::CallArgs args = JS::CallArgsFromVp(argc, vp);
-    fprintf(stdout, "built on %s at %s\n", __DATE__, __TIME__);
-    args.rval().setUndefined();
-    return true;
-}
-
-static bool
 Quit(JSContext *cx,
      unsigned argc,
      JS::Value *vp)
@@ -275,7 +266,6 @@ const JSFunctionSpec gGlobalFunctions[] =
     JS_FS("load",            Load,           1,0),
     JS_FS("quit",            Quit,           0,0),
     JS_FS("version",         Version,        1,0),
-    JS_FS("build",           BuildDate,      0,0),
     JS_FS("dumpXPC",         DumpXPC,        1,0),
     JS_FS("dump",            Dump,           1,0),
     JS_FS("gc",              GC,             0,0),
diff --git a/mozilla/js/src/shell/js.cpp b/mozilla/js/src/shell/js.cpp
index 4e9a589..032bfa3 100644
--- a/mozilla/js/src/shell/js.cpp
+++ b/mozilla/js/src/shell/js.cpp
@@ -2521,15 +2521,6 @@ DisassWithSrc(JSContext* cx, unsigned argc, jsval* vp)
 #endif /* DEBUG */
 
 static bool
-BuildDate(JSContext* cx, unsigned argc, jsval* vp)
-{
-    CallArgs args = CallArgsFromVp(argc, vp);
-    fprintf(gOutFile, "built on %s at %s\n", __DATE__, __TIME__);
-    args.rval().setUndefined();
-    return true;
-}
-
-static bool
 Intern(JSContext* cx, unsigned argc, jsval* vp)
 {
     CallArgs args = CallArgsFromVp(argc, vp);
@@ -4857,10 +4848,6 @@ static const JSFunctionSpecWithHelp shell_functions[] = {
 "  Similar to the DumpJSStack() function in the browser."),
 
 #endif
-    JS_FN_HELP("build", BuildDate, 0, 0,
-"build()",
-"  Show build date and time."),
-
     JS_FN_HELP("intern", Intern, 1, 0,
 "intern(str)",
 "  Internalize str in the atom table."),
diff --git a/mozilla/js/xpconnect/src/XPCShellImpl.cpp b/mozilla/js/xpconnect/src/XPCShellImpl.cpp
index fc13929..14b9a80 100644
--- a/mozilla/js/xpconnect/src/XPCShellImpl.cpp
+++ b/mozilla/js/xpconnect/src/XPCShellImpl.cpp
@@ -373,15 +373,6 @@ Version(JSContext* cx, unsigned argc, jsval* vp)
 }
 
 static bool
-BuildDate(JSContext* cx, unsigned argc, jsval* vp)
-{
-    CallArgs args = CallArgsFromVp(argc, vp);
-    fprintf(gOutFile, "built on %s at %s\n", __DATE__, __TIME__);
-    args.rval().setUndefined();
-    return true;
-}
-
-static bool
 Quit(JSContext* cx, unsigned argc, jsval* vp)
 {
     CallArgs args = CallArgsFromVp(argc, vp);
@@ -669,7 +660,6 @@ static const JSFunctionSpec glob_functions[] = {
     JS_FS("quit",            Quit,           0,0),
     JS_FS("ignoreReportedErrors", IgnoreReportedErrors, 1,0),
     JS_FS("version",         Version,        1,0),
-    JS_FS("build",           BuildDate,      0,0),
     JS_FS("dumpXPC",         DumpXPC,        1,0),
     JS_FS("dump",            Dump,           1,0),
     JS_FS("gc",              GC,             0,0),
